lib/kennel/models/record.rb

```diff
@@ -179,8 +225,14 @@ def resolve_link(sought_tracking_id, sought_type, id_map, force:)
       end
 
       # let users know which project/resource failed when something happens during diffing where the backtrace is hidden
-      def invalid!(message)
-        raise ValidationError, "#{tracking_id} #{message}"
+      def invalid!(tag, message = nil)
+        if message.nil?
```

Comment:
maybe some todo comment here to clean this up later

Reply:
In my current sketch of the branch, there are 7 possible ways that a message can be reported; it's quite a complex set of `if`s. Once we've completed the migration from `invalid` to `skip_validations`, it would be reduced to 4.